planner/core/rule_partition_processor.go

```diff
@@ -126,17 +122,37 @@ func (s *partitionProcessor) prune(ds *DataSource) (LogicalPlan, error) {
 	return unionAll, nil
 }
 
-// canBePrune checks if partition expression will never meets the selection condition.
+var solver = expression.NewPartitionPruneSolver()
+
+// canBePruned checks if partition expression will never meets the selection condition.
 // For example, partition by column a > 3, and select condition is a < 3, then canBePrune returns true.
-func (s *partitionProcessor) canBePrune(ctx sessionctx.Context, col *expression.Column, partitionCond expression.Expression, copConds []expression.Expression) (bool, error) {
-	conds := make([]expression.Expression, 0, 1+len(copConds))
-	conds = append(conds, partitionCond)
-	conds = append(conds, copConds...)
-	conds = expression.PropagateConstant(ctx, conds)
+func (s *partitionProcessor) canBePruned(sctx sessionctx.Context, partCol *expression.Column, partitionExpr expression.Expression, filterExprs []expression.Expression) (bool, error) {
+	conds := make([]expression.Expression, 0, 1+len(filterExprs))
+	conds = append(conds, partitionExpr)
+	conds = append(conds, filterExprs...)
+	conds = expression.PropagateConstant(sctx, conds)
+	conds = solver.Solve(sctx, conds)
+
+	if len(conds) == 1 {
```

Comment:
We don't need to calculate ranges when `len(conds) = 1` but it's not a constant?